src/Features/CSharp/Portable/UseNamedArguments/CSharpUseNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -30,6 +33,22 @@ protected sealed override bool IsCloseParenOrComma(SyntaxToken token)
 
             protected override bool SupportsNonTrailingNamedArguments(ParseOptions options)
                 => ((CSharpParseOptions)options).LanguageVersion >= LanguageVersion.CSharp7_2;
+
+            protected override bool IsImplicitIndexOrRangeIndexer(ImmutableArray<IParameterSymbol> parameters, TSyntax argument, SemanticModel semanticModel)
+            {
+                var argType = semanticModel.GetTypeInfo(GetArgumentExpression(argument)).Type;
+                if (argType?.ContainingNamespace is { Name: nameof(System), ContainingNamespace: { IsGlobalNamespace: true } } &&
+                    (argType.Name == "Range" || argType.Name == "Index"))
+                {
+                    var conversion = semanticModel.Compilation.ClassifyConversion(argType, parameters[0].Type);
+                    if (!conversion.Exists || conversion.IsExplicit)
```

Comment:
not quite understanding this check.

Reply:
@ryzengard, @cyrusNajmabadi

I can't think of any obvious way to tell if an implicit range or index indexer was used.

The best heuristic I could come up with is to check if the parameter doesn't fit the method it's being used with. The easiest way to check that is to see if it only has at most an explicit conversion to the indexers parameter types.

For range I could check if the number of parameters is correct. That won't work for index though.

For index I could check if the parameter is an integer, but maybe your definition of Range defines an implicit conversion to int, and so it can actually be used with a normal indexer.

@ryzengard, with regards to your example, I'm not sure which method overload resolution prefers, but whichever it is, that is the method symbol we will check against, and so my check should work correctly.

Reply:
i would prefer we fix this up to fit the expected usage of using the actual Systme.Range/Index.  We should also document this code tomake it clear what it's doing for the next person who comes along and has to fix it up.  Thanks!

Reply:
> i would prefer we fix this up to fit the expected usage of using the actual Systme.Range/Index.

I'm not sure what you mean by this?

Reply:
Comment Added @CyrusNajmabadi